server/stream.go

```diff
@@ -613,6 +724,21 @@ func (mset *Stream) processInboundJetStreamMsg(_ *subscription, pc *client, subj
 	name := mset.config.Name
 	maxMsgSize := int(mset.config.MaxMsgSize)
 	numConsumers := len(mset.consumers)
+
+	// Process msgId if we have headers.
+	var msgId string
+	if pc != nil && pc.pa.hdr > 0 {
+		msgId = getMsgId(msg[:pc.pa.hdr])
```

Comment:
Having headers does not guarantee that the JSPubId will be present, so should check msg Id for being say empty to indicate that there is none.

Reply:
msgId will be "" and the rest of the code will do the right thing.

Reply:
Sure, but if we could save a map lookup, why not?

Reply:
We don't do a map lookup if we have an empty key.